### ISSUE TITLE
Not working in Mac OS with M1 Chip (solution inside)

### ISSUE THREAD

**Issue by LaiKash** [bkerler/mtkclient#1064]
I managed to get it works. Please feel free to close this issue, this is for future reference or to add it to the readme if you please.

## Installation in MacOs with M1 chip

I am using v1.6.3 in my personal case as the preloader injection for camellia doesn't work in the latest version, but should be the same.

Run a python 3.9 venv: 

```
python3.9 -m venv mtk_venv
source mtk_venv/bin/activate
```

For MacOs we need some dependencies (as I said better inside a venv):

```
brew install macFUSE 
pip3 install --pre --no-binary capstone capstone
pip3 install PySide6
pip3 install libusb
```

Then install it normally as in the Github says (or build it):

```
pip3 install -r requirements.txt
pip3 install .
```

If you get a Open SSL issue add the following to your .zshrc file and source it afterwards (you might need to edit the path of openssl depending on how you installed it):

```bash
export PATH="/opt/homebrew/bin:$PATH"
export CFLAGS="-I/opt/homebrew/opt/openssl/include"
export LDFLAGS="-L/opt/homebrew/opt/openssl/lib"
```

To send the preloader it is a bit tricky. Many issues and commits ago, Kamakiri (not Kamakiri2) was hardcoded for MacOs. For me it is the oposite, works with kamakiri2, not with normal kamakiri.

 You must run it with sudo and use kamakiri2 as payload or it fails:

`sudo python3 ./mtk plstage --ptype="kamakiri2" --preloader preloader_camellia.bin`

**Comment by 00p513-dev** [comment]
Which version of macOS did you try on? I will test for myself :)

**Comment by LaiKash** [comment]
> Which version of macOS did you try on? I will test for myself :)

Ventura, 13.4.1 :D

**Comment by bkerler** [comment]
So I assume this issue can be closed as #1083 is merged ?

**Comment by bkerler** [comment]
Ok, closing it now.
